src/sandbox.rkt

```diff
@@ -159,22 +162,21 @@
                       repr)))
   (timeline-push! 'bogosity domain-stats)
   (define-values (train-pcontext test-pcontext)
-    (split-pcontext joint-pcontext (*num-points*) (*reeval-pts*))) 
-
-  (define end-alts
-    (run-improve! (test-input test) train-pcontext (*num-iterations*)
-                  #:specification (test-spec test)
-                  #:preprocess (test-preprocess test)))
-
+    (split-pcontext joint-pcontext (*num-points*) (*reeval-pts*)))
+  ;; TODO: Ignoring all user-provided preprocessing right now
+  (define-values (end-alts preprocessing)
+    (run-improve!
+     (or (test-spec test) (test-input test)) ctx train-pcontext (*simplify-rules*)
```

Comment:
Wait, why are we passing in the `test-spec` here?

Reply:
It's kind of confusing! If you go check out main, `run-improve!` took a `#:specification` keyword argument, and `get-altneratives/report` always passes that in with `(test-spec test)`. There was a comment mentioning that we use the specification for sampling points which I forgot to duplicate when I moved things around, but I just added it back.

https://github.com/herbie-fp/herbie/blob/main/src/mainloop.rkt#L355
https://github.com/herbie-fp/herbie/blob/main/src/sandbox.rkt#L166

I also remember I asked you about this in the Herbie channel a while back:

https://usocgrad.slack.com/archives/C0G44MH4P/p1687299376132259

Reply:
Yeah the issue here is that if there is a specification, the input program is never being passed in at all! This is definitely wrong.